iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesParser.java

```diff
@@ -63,4 +74,34 @@ public static boolean hasHelmContent(String text) {
     }
     return false;
   }
+
+  @Nullable
+  String evaluateHelmTemplate(String filename, String source, InputFileContext inputFileContext) {
+    // TODO: better support of Helm project structure
+    var valuesFile = findValuesFile(inputFileContext);
+    if (valuesFile != null) {
+      try {
+        return helmPreprocessor.evaluateTemplate(filename, source, valuesFile.contents());
+      } catch (IOException e) {
+        LOG.debug("Failed to read values file '{}', skipping processing of Helm file '{}'", valuesFile.filename(), filename, e);
+      }
+    } else {
+      LOG.debug("Failed to read values file, skipping processing of Helm file '{}'", filename);
+    }
+
+    return null;
+  }
+
+  static @Nullable InputFile findValuesFile(InputFileContext inputFileContext) {
+    var valuesFilePredicate = valuesFilePredicate(inputFileContext.sensorContext);
+    return inputFileContext.sensorContext.fileSystem().inputFile(valuesFilePredicate);
+  }
+
+  private static FilePredicate valuesFilePredicate(SensorContext sensorContext) {
+    FilePredicates predicates = sensorContext.fileSystem().predicates();
+    return predicates.and(
+      predicates.hasLanguage("yaml"),
+      predicates.hasType(InputFile.Type.MAIN),
+      predicates.hasFilename("values.yaml"));
```

Comment:
IMO here we should try to load the `../values.yaml` only (the relative path to template file). In case the repo contains multiple Charts this will return multiple values.yaml from other Charts.